File: backend/schemas/opportunities.py

import uuid
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class OpportunityCreate(BaseModel):
    title: str
    description: str
    opportunity_type_id: uuid.UUID
    company: str
    location: Optional[str] = None
    deadline: Optional[datetime] = None

class OpportunityResponse(BaseModel):
    id: uuid.UUID
    title: str
    description: str
    opportunity_type_id: uuid.UUID
    company: str
    location: Optional[str]
    deadline: Optional[datetime]
    status: str
    created_by: uuid.UUID
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/services/opportunities.ts
import { apiClient } from '../client';

export interface Opportunity {
  id: string;
  title: string;
  description: string;
  company: string;
  location: string | null;
  deadline: string | null;
  status: string;
  created_at: string;
  created_by: string;
}

export const getOpportunities = async (params?: { search?: string, company?: string, status?: string }) => {
  const data: any = await apiClient.get('/opportunities/', { params });
  return data.data as Opportunity[];
};
